### PR TITLE
Adds confirmation dialog upon clicking the trash icon in the ledger table.

### PR DESCRIPTION
**Pull request by ryanml**
Fixes Issue #11164

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [x] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:

 1. Visit some sites
 2. Open about:payments
 3. Click a trash icon
 4. Ensure that the appearing confirmation dialogue, depending on the choice, allows the removal of the site or does nothing. 

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [x] Adequate test coverage exists to prevent regressions
- [x] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

Two translations added under the name: banSiteConfirmation. Mechanism for creating the confirmation dialogue mimics that shown in the syncTable component. Ref: [here](https://github.com/brave/browser-laptop/blob/master/app/renderer/components/preferences/syncTab.js#L392)

### CODE REVIEW

**Comment by ryanml** [comment]
@bsclifton I hope I got the PR title right this time ;)

I hope that the dialogue provided in this PR is appropriate, should a change in language work better, let me know and I can update those locales.

cc: @NejcZdovc

**Comment by bsclifton** [comment]
@bradleyrichter how does the wording for this look?
`Are you sure you want to delete this site?`

screen shot 2018-02-16 at 11 24 11 am

**Comment by bsclifton** [comment]
Will review of this up to @NejcZdovc 😄 There are some planned changes for this UI (we may want to make those first)

File: app/extensions/brave/locales/es/preferences.properties [hunk 1]
```diff
@@ -24,6 +24,7 @@ autoplay=Autoplay Media
 autoSuggestSites=auto-incluir
 backupLedger=Copia de seguridad de tu cartera
 balanceRecovered=Se ha recuperado un total de {{balance}} y se han transferido a tu cartera Brave.
+banSiteConfirmation=¿Seguro que quieres eliminar este sitio?
```
Comment (NejcZdovc) [inline app/extensions/brave/locales/es/preferences.properties]: Please remove this line. We are doing translations via Transifex https://github.com/brave/browser-laptop/blob/master/docs/translations.md

Comment (ryanml) [reply app/extensions/brave/locales/es/preferences.properties]: @NejcZdovc apologies there, i've removed this line.

**Comment by NejcZdovc** [comment]
@ryanml can you please squash commits? Then we are good to go

**Comment by ryanml** [comment]
See #13182

**Pull request closed without merging**